content/en/blog/_posts/2025-04-17-volume-populators-ga/index.md

```diff
@@ -0,0 +1,73 @@
+    apiGroup: provider.example.com
+    kind: Provider
+    name: provider1
+```
+
+## What is new
+
+There are four major enhancements from beta.
+
+### Populator Pod is optional
+
+During the beta phase, we discovered that initiating data population doesn't always require a dedicated populator pod. In some cases, it's more efficient to interact with the provider's APIs directly from the controller pod.
+
+To accommodate this, we've introduced three new plugin-based functions:
+* PopulateFn(): Executes the provider-specific data population logic.
+* PopulateCompleteFn(): Checks if the data population operation has finished successfully.
+* PopulateCleanupFn(): Cleans up temporary resources created by the provider-specific functions after data population is completed
+
+A provider example is added in [lib-volume-populator/example](https://github.com/kubernetes-csi/lib-volume-populator/tree/master/example).
+
+### Mutator functions to modify the Kubernetes resources
+
+For GA a MutatorConfig to the Volume Populator Controller is added, allowing the specification of mutator functions to modify Kubernetes resources. For example, if the PVC prime is not an exact copy of the PVC and you need provider-specific information for the driver, you can include this information in the optional MutatorConfig. This allows you to customize the default Kubernetes objects in the volume populator.
+
+### Flexible metric handling for providers
+
+Our beta phase highlighted a new requirement: the need to aggregate metrics not just from lib-volume-populator, but also from other components within the provider's codebase.
+
+To address this, we've introduced a [provider metric manager](https://github.com/kubernetes-csi/lib-volume-populator/blob/8a922a5302fdba13a6c27328ee50e5396940214b/populator-machinery/controller.go#L122). This enhancement delegates the implementation of metrics logic to the provider itself, rather than relying solely on lib-volume-populator. This shift provides greater flexibility and control over metrics collection and aggregation, enabling a more comprehensive view of provider performance.
+
+### Clean up temp resources
+
+During the beta phase, we identified potential resource leaks with PersistentVolumeClaim (PVC) deletion while volume population was in progress, due to limitations in finalizer handling. We have improved the populator to support the deletion of temporary resources (PVC prime, etc.) if the original PVC is deleted in this GA release.
+
+## How to use it
+
+To try it out, please follow the [steps](/blog/2022/05/16/volume-populators-beta/#trying-it-out) in the previous beta blog.
+
+## Future feature requests
+
+For next step, there are several potential feature requests for volume populator:
+
+* Multi sync: the current implementation is a one-time unidirectional sync from source to destination. This can be extended to support multiple syncs, which opens up an opportunity to run periodic sync, or allow users to sync on demand
```

Comment:
Sorry really minor one but noticed a grammar error

```suggestion
* Multi sync: the current implementation is a one-time unidirectional sync from source to destination. This can be extended to support multiple syncs, enabling periodic syncs or allowing users to sync on demand
```

Reply:
Updated